telegram/base.py

```diff
@@ -80,3 +80,25 @@ def to_dict(self):
                     data[key] = value
 
         return data
+
+    def _get_id(self):
+        for x in ['id', 'file_id', 'message_id', 'result_id', 'update_id']:
+            if hasattr(self, x):
+                return self.__getattribute__(x)
+        raise NotImplementedError
```

Comment:
I return NotImplementedError as a way of saying "this object doesn't have an id", which many objects do not (pretty much all objects not mentioned in the OP). This is why I'm catching it inside of `__eq__` too.